nccl4py/pyproject.toml

```diff
@@ -0,0 +1,63 @@
+[project]
+name = "nccl4py"
+dynamic = ["version", "dependencies"]
+requires-python = ">=3.10"
+description = "Python bindings for NCCL"
+authors = [
+    { name = "NVIDIA Corporation" }
+]
+readme = "README.md"
+
+[project.optional-dependencies]
+dev = [
+    "pytest>=6.0",
+    "pytest-cov",
+    "pytest-mpi",
+    "mpi4py",
+    "ruff",
+    "pyright",
+    "ml-dtypes",
+]
+
+[build-system]
+requires = [
+    "setuptools>=64",
+    "setuptools_scm",
+    "wheel",
+    "Cython>=0.29.24"
+]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools]
+include-package-data = true
+
+[tool.setuptools.packages.find]
+include = ["nccl", "nccl.*"]
+exclude = ["__pycache__"]
+
+[tool.setuptools.package-data]
+"nccl.bindings" = ["*.pyi"]
+
+[tool.setuptools.exclude-package-data]
+"*" = ["__pycache__/*", "*.py[co]", "*.stamp", "*.pyx", "*.pxd", "*.cpp", "*.c"]
```

Comment:
All *.pxd files are excluded. However, things like `cynccl.pxd` may still be useful for third party code, as it provides access to the core NCCL API. Maybe this file should be promoted to the top-level package directory and exposed to third-party Cython code? That's what I do in mpi4py with `libmpi.pxd`, though I have no idea how much people use it.

Reply:
Yes. By design, `cynccl.pxd` is meant to be public facing. However, I don't think this is currently tested by nccl4py? If not, then we should document that the Cython interface is experimental.